### DISCUSSION TITLE
Bisq2-web-ui Milestone plan for MVP

### DISCUSSION THREAD

**Discussion by salokiam** [bisq-network/bisq2#1312] [General]
The Bisq2-web-ui project is a subproject inside Bisq2 (which has natively a desktop UI build with JavaFX). It's intended to be run on Personal Nodes like Start9, Umbrel, MyNode,... See also https://github.com/bisq-network/proposals/issues/403

In the following I want to lay out the further steps for this project and ask for feedback. The next step for this Project would be to develop an MVP (minimal viable product). How does the MVP look like? I took the existing Bisq2 desktop app, made screenshots from it and crossed out in red all parts that I intend to not include in the web UI for this release. I think the shown subset is complete in a sense that an user could work with it. (although not having all features from the web UI) see [Bisq2-reduktion2web-v3.pdf](https://github.com/bisq-network/bisq2/files/13309999/Bisq2-reduktion2web-v3.pdf)
Milestones (details need to be worked out as we go):

## Milestone update
- Java 17
- Vaadin version 23→24
- update Bisq2 to latest version

***Timeline: November 23, estimated cost: $6500***

## Milestone chat

- offerbook ('create offer'-wizard will be made later, see below)

- private chats

***Timeline: Januar 24, estimated cost: $8000***

## Milestone open trades
- create offer wizard
- open trades

***Timeline: March 24, estimated cost: $17000***
[Bisq2-reduktion2web-v3.pdf](https://github.com/bisq-network/bisq2/files/13309999/Bisq2-reduktion2web-v3.pdf)

**Comment by HenrikJannsen** [comment]
Timeline for 3rd package is different in pdf and GH.

**Reply by salokiam** [reply]
fixed it.

**Comment by HenrikJannsen** [comment]
I think left nav can be dropped and `User` implemented as icon at top bar. `Settings` if needed as well. IMO `User` could be dropped also, not strictly needed for a MVP. Same for Payment accounts.

**Reply by salokiam** [reply]
> I think left nav can be dropped and User implemented as icon at top bar. Settings if needed as well.

I agree


>IMO User could be dropped also, not strictly needed for a MVP. Same for Payment accounts.
 

If we drop 'User Profile'
1. User Profile, then there is only one profile (which is ok, for the beginning)
2. User->Payment account, then the user needs to enter his Account data each time when selling. (ok, with me)
3. User->password User should have ability to change the password. This is  not a large dialog anyway, so I would vote for keeping it.

So we could drop User expect for password.


Dropping 'Settings' would mean:
1. no language selection, so user is stuck with english. (fine for me for first release)
2. User can't make a backup. Since this is a chat anyway a backup doesn't make much sense anyway.
3. 'Open log file', I guess we need that, because if installed on a personal server like Umbrel its difficult to get that in case the user wants to report a bug.

So we could drop Setting, except for 'Open log file'.

I will update the PDF. Lets see if there are more suggestions...

**Reply by HenrikJannsen** [reply]
> User->password User should have ability to change the password. This is not a large dialog anyway, so I would vote for keeping it.

But comes with more strings attached (password entering, data encryption). as we dont have a wallet yet i would consider it not much important.

Setting should not be dropped. But backup and logs I dont consider must haves. Lets reduce all to the minimum, its enough effort to get to the state of current Bisq Easy.

Can you add cost estimations as well?

**Reply by salokiam** [reply]
> Lets reduce all to the minimum,

ok, reduced it further for the MVP.
I also added cost estimation.

**Reply by salokiam** [reply]
and reduced timeframe by one month

**Comment by salokiam** [comment]
updated the PDF and original post

**Comment by EmmanuelEklipse** [comment]
How would the hosting of the website be done to ensure that it's decentralized?

**Reply by salokiam** [reply]
Everybody would have their own personel server hosted on a Umbrel, Start9 or MyNode. So each webapp would have exactly one user.

**Comment by HenrikJannsen** [comment]
Close as put on hold as for Bisq Easy there is no need to leave a node running as offers stay online.
